splinepy/helpme/extract.py

```diff
@@ -9,102 +10,99 @@
+            closed=False,
+        )
+        if isinstance(spline, Multipatch):
+            e = np.vstack(
+                [e + i * resolution for i in range(len(spline.splines))]
+            )
         return Edges(
-            vertices=spline.sample(resolution),
-            edges=connec.range_to_edges(
-                (0, resolution),
-                closed=False,
-            ),
+            vertices=vertices,
+            edges=e,
         )
 
     else:
-        # This should be possible for spline of any dimension.
-        # As long as it satisfies the following condition
-        if extract_knot is not None:
-            if len(extract_knot) != spline.para_dim - 1:
-                raise ValueError(
-                    "Must satisfy len(extract_knot) == spline.para_dim -1."
-                )
+        # recursion for Multipatch splines
+        if isinstance(spline, Multipatch):
+            return Edges.concat(
+                [
+                    edges(spline=s, resolution=resolution, all_knots=all_knots)
+                    for s in spline.splines
+                ]
+            )
 
-        # This may take awhile.
+        # Single patch case
         if all_knots:
-            temp_edges = []  # edges' is not a valid syntax
-            unique_knots = np.array(spline.unique_knots, dtype=object)
-            for i in range(spline.para_dim):
-                mask = np.ones(spline.para_dim, dtype=bool)
-                mask[i] = False
-                # gather knots along current knot
-                extract_knot_queries = cartesian_product(
-                    unique_knots[mask], reverse=False
-                )
-
-                for ekq in extract_knot_queries:
-                    temp_edges.append(
-                        edges(spline, resolution[i], i, ekq, False)
-                    )
-
-            return Edges.concat(temp_edges)
-
-        # Get parametric points to extract
-        queries = np.empty(
-            (resolution, spline.para_dim),
-            dtype="float64",  # hardcoded for splinelibpy
-            order="C",  # hardcoded for splinelibpy
-        )
-        # get ~extract_dim
-        not_ed = np.arange(spline.para_dim).tolist()
-        not_ed.pop(extract_dim)
-        queries[:, not_ed] = extract_knot
-
-        # get knot extrema
-        uniq_knots = spline.unique_knots[extract_dim]
-        min_knot_position = min(uniq_knots)
-        max_knot_position = max(uniq_knots)
-
-        queries[:, extract_dim] = np.linspace(
-            min_knot_position,
-            max_knot_position,
-            resolution,
-        )
+            relevant_knots = spline.unique_knots
+        else:
+            relevant_knots = list(spline.parametric_bounds.T)
 
-        return Edges(
-            vertices=spline.evaluate(queries),
-            edges=connec.range_to_edges(
+        temp_edges = []  # edges' is not a valid syntax
+        for i in range(spline.para_dim):
+            reorder_mask = (
```

Comment:
@jzwar some comments for future selves here?

Reply:
done